geometry/parametrized_segment.hpp

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "geometry/point2d.hpp"
+
+#include "base/math.hpp"
+
+#include <cmath>
+#include <limits>
+
+namespace m2
+{
+// This class holds a parametrization of the
+// line segment between two points p0 and p1.
+// The parametrization is of the form
+//   p(t) = p0 + t * dir.
+// Other conditions:
+//   dir is the normalized (p1 - p0) vector.
+//   length(dir) = 1.
+//   p(0) = p0.
+//   p(T) = p1 with T = length(p1 - p0).
+//
+// The points with t in [0, T] are the points of the segment.
+template <typename Point>
+class ParametrizedSegment
+{
+public:
+  static_assert(std::numeric_limits<typename Point::value_type>::is_signed,
+                "Unsigned points are not supported");
+
+  ParametrizedSegment(Point const & p0, Point const & p1) : m_p0(p0), m_p1(p1)
+  {
+    m_d = m_p1 - m_p0;
+    m_length = std::sqrt(m_d.SquaredLength());
+    if (m_d.IsAlmostZero())
+      m_d = Point::Zero();
+    else
+      m_d = m_d / m_length;
+  }
+
+  double SquaredDistanceToPoint(Point const & p) const
```

Comment:
Probably it's worth adding a short comments for these two methods of the class.

Reply:
done